### PR TITLE
Adding support viatable with tests

### PR DESCRIPTION
**Pull request by nixmen**
- fixed unit-tests
- adding supporting relations with viaTable
- fix php version for Travis CI

### CODE REVIEW

File: RelationBehavior.php [hunk 1]
```diff
@@ -160,13 +162,17 @@ protected function loadData()
             $getter = 'get' . ucfirst($attribute);
             $data['activeQuery'] = $activeQuery = $this->owner->$getter();
             $data['newModels'] = [];
+            $data['oldModels'] = [];
```
Comment (senor-artemisio) [inline RelationBehavior.php:164]: its missed in PhpDoc

File: README.md [hunk 2]
```diff
@@ -171,4 +172,51 @@ class Entity extends ActiveRecord
...
+        return [
+            [['many_to_many_attribute'], 'safe']
+        ];
+    }
+
+    public function getMany_to_many_attribute()
+    {
+        return $this->hasMany(ManyToManyModel::className(), 
+            ['id' => 'many_to_many_model_id'])
+                ->viaTable('table_many_to_many', ['entity_id' => 'id']);
```
Comment (senor-artemisio) [inline README.md:199]: missing example with on condition for join with table_many_to_many table

Comment (senor-artemisio) [reply README.md:199]: I use such relation


```php
<?php
return $this->hasMany(Woo::className(), ['id' => 'woo_id'])
   ->viaTable(WooRelation::tableName(), ['item_id' => 'id'], function ($query) {
     /** @var ActiveQuery $query */
     return $query->onCondition(['item_type' => 'woo']);
   }
);
```

and it's didn't work fine - old tags removes after save, remains only new tags.

If problem in RelationBehavior, you shoud fix it, else just add correct example that will works.

File: tests/unit/FakeNewsModel.php [hunk 3]
```diff
@@ -86,6 +93,15 @@ public function getNews_files()
         return $this->hasMany(FakeFilesModel::className(), ['id' => 'file_id'])->via('newsFiles');
     }
 
+    /**
+     * @return $this
+     */
+    public function getNews_files_via_table()
+    {
+        return $this->hasMany(FakeFilesModel::className(), ['id' => 'file_id'])->viaTable('news_files_via_table',
+            ['news_id' => 'id']);
```
Comment (senor-artemisio) [inline tests/unit/FakeNewsModel.php:121]: missing test with on condition

**Review by senor-artemisio — CHANGES_REQUESTED** [review]
In general, everything is fine, but you need to solve the problem with on condition

File: README.md (hunk 2 above)
Comment (nixmen) [reply README.md:199]: fixed

File: tests/unit/FakeNewsModel.php (hunk 3 above)
Comment (nixmen) [reply tests/unit/FakeNewsModel.php:121]: add

File: tests/unit/RelationBehaviorTest.php [hunk 4]
```diff
@@ -465,7 +607,586 @@ public function testIsDeletedModel()
...
+        $method = new \ReflectionMethod(
+            $behavior, 'loadModelsManyToManyVia'
+        );
+        $method->setAccessible(true);
+        $method->invokeArgs($behavior, ['news_files_w_cond']);
+
+        $this->assertEquals($newModelsExpected, $prop->getValue($behavior)['news_files_w_cond']['newModels']);
+        $this->assertEquals($oldModelsExpected, $prop->getValue($behavior)['news_files_w_cond']['oldModels']);
+
+        die('end');
```
Comment (senor-artemisio) [inline tests/unit/RelationBehaviorTest.php]: WTF?!

File: tests/unit/FakeNewsModel.php [hunk 5]
```diff
@@ -7,6 +7,13 @@
 
 /**
  * Тестовая модель Новости
```
Comment (senor-artemisio) [inline tests/unit/FakeNewsModel.php]: it's should be translated to English

File: tests/unit/FakeNewsModel.php [hunk 6]
```diff
@@ -79,13 +86,53 @@ public function getNewsFiles()
...
+     */
+    public function getNewsFilesWithCond()
+    {
+        return $this
+            ->hasMany(FakeNewsFilesModel::className(), ['news_id' => 'id'])
+            ->onCondition(['entity_type' => 'with_condition']);
+    }
+
+    /**
      * @return $this
```
Comment (senor-artemisio) [inline tests/unit/FakeNewsModel.php]: wrong PHPDoc

**Review by senor-artemisio — CHANGES_REQUESTED** [review]
why file names is duplicated?

File: tests/unit/FakeNewsModel.php (hunk 6 above)
Comment (nixmen) [reply tests/unit/FakeNewsModel.php]: fixed

File: tests/unit/FakeNewsModel.php (hunk 5 above)
Comment (nixmen) [reply tests/unit/FakeNewsModel.php]: translated

File: tests/unit/RelationBehaviorTest.php (hunk 4 above)
Comment (nixmen) [reply tests/unit/RelationBehaviorTest.php]: fixed

**Pull request closed without merging**